journal.py

```diff
@@ -69,27 +72,43 @@ def read_entry(request):
     return {'entry': entry}
 
 
-@view_config(route_name='edit', renderer='templates/edit.jinja2')
+@view_config(route_name='edit', renderer='json')
 def edit_entry_view(request):
     """return a list of all entries as dicts"""
     if request.authenticated_userid:
-        cursor = request.db.cursor()
-        cursor.execute(DB_ENTRY, (request.matchdict['id'], ))
-        keys = ('id', 'title', 'text', 'created')
-        entries = [dict(zip(keys, row)) for row in cursor.fetchall()]
-
-        if request.method == 'POST':
+        if request.method == 'GET':
+            # import pdb; pdb.set_trace();
+            cursor = request.db.cursor()
+            cursor.execute(DB_ENTRY, (request.params.get('id', None), ))
+            keys = ('id', 'title', 'text', 'created')
+            row = cursor.fetchone()
+            entry = dict(zip(keys, row))
+            entry['created'] = entry['created'].strftime('%b %d, %Y')
+
+            return entry
+
+        elif request.method == 'POST':
+            # import pdb; pdb.set_trace()
             try:
                 edit_entry(request)
             except psycopg2.Error:
                 # this will catch any errors generated by the database
-                return HTTPInternalServerError
-            return HTTPFound(request.route_url('home'))
+                return HTTPInternalServerError()
+
+            cursor = request.db.cursor()
+            cursor.execute(DB_ENTRY, (request.params.get('id', None), ))
+            keys = ('id', 'title', 'text', 'created')
+            # import pdb; pdb.set_trace()
+            row = cursor.fetchone()
+            entry = dict(zip(keys, row))
```

Comment:
Notice how you repeat the code for retrieving the entry from the database and turning it into a dict in two places.  Is that really something you have to do?  Could you do it just once?